http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -1016,10 +967,15 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
         else
           Right(allLabels)
 
-      // TODO make this non optional in the request
-      // the original thought when developing was that galaxy needs a disk, but some apps may not
-      // all apps require a disk
-      disk <- diskOpt.toRight(AppRequiresDiskException(cloudContext, appName, req.appType, ctx.traceId))
+      // Galaxy, Cromwell, and Custom app types require a disk.
+      // Cromwell on Azure requires _no_ disk.
+      _ <- (req.appType, diskOpt) match {
+        case (Galaxy | Cromwell | Custom, None) =>
+          Left(AppRequiresDiskException(cloudContext, appName, req.appType, ctx.traceId))
+        case (CromwellOnAzure, Some(_)) =>
+          Left(AppDiskNotSupportedException(cloudContext, appName, req.appType, ctx.traceId))
```

Comment:
Disk is not supported for Azure apps -- the API now returns 400 if the user tries to specify one.